### DISCUSSION TITLE
call: no common audio codecs - INVITE through a WS to UDP bridge

### DISCUSSION THREAD

**Discussion by eafunk** [baresip/baresip#3075] [Q&A]
I am trying to get baresip to accept a call from a web client based on jssip-3.10.0, which communicated SIP messages with baresip via a Websocket (WS) to UDP bridge program I created with node.js.  This is not working with baresip rejecting the call error "no common audio codecs," even though I don't see anything wrong with the SIP message conversation going through my bridge program.

baresip accounts config file contents:
<sip:51@192.168.15.5>;auth_pass=xxxxxxxxxxxxxxxxxx;answermode=auto
<sip:peer@192.168.15.3;transport=ws>;regint=0;answermode=auto

Here is baresip running with console output.  Registration, and call answering (not shown here), both work fine from asterisk, but fails as seen on the Second to last line call rejection, when call is placed from jssip-3.10.0 web client, placed through a WS to UDP bridge program I created in node.js.  All this happens inside a local network where every program involved can see everyone else.  No NAT, etc.
$ baresip -v
baresip v1.0.0 Copyright (C) 2010 - 2020 Alfred E. Heggestad et al.
Local network address:  IPv4=enp1s0|192.168.15.3 
ui: stdio
cons: UI console listening on 0.0.0.0:5555
ui: cons
aucodec: PCMU/8000/1
aucodec: PCMA/8000/1
gsm: GSM v1.0.22
aucodec: GSM/8000/1
aucodec: L16/48000/2
aucodec: L16/44100/2
aucodec: L16/32000/2
aucodec: L16/16000/2
aucodec: L16/8000/2
aucodec: L16/48000/1
aucodec: L16/44100/1
aucodec: L16/32000/1
aucodec: L16/16000/1
aucodec: L16/8000/1
auplay: jack
ausrc: jack
vidcodec: H264
vidcodec: H264
vidcodec: H263
vidcodec: MP4V-ES
vidcodec: H265
avcodec: using H.264 encoder 'libx264' -- libx264 H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
avcodec: using H.264 decoder 'h264' -- H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
avcodec: using H.265 encoder 'libx265' -- libx265 H.265 / HEVC
avcodec: using H.265 decoder 'hevc' -- HEVC (High Efficiency Video Coding)
mediaenc: srtp
mediaenc: srtp-mand
mediaenc: srtp-mandf
uuid: loaded UUID 58d584a9-25b9-d597-02c0-91f439e8a4c5 from file /home/bcaster/.baresip/uuid
Populated 2 accounts
module: loading app auloop.so
module: loading app contact.so
Populated 3 contacts
module: loading app ctrl_tcp.so
ctrl_tcp: TCP socket listening on 0.0.0.0:4444
module: loading app vidloop.so
Populated 13 audio codecs
Populated 0 audio filters
Populated 5 video codecs
Populated 0 video filters
baresip is ready.
51@192.168.15.5: {0/UDP/v4} 200 OK (Asterisk PBX 13.13.1) [1 binding]
ua: sipsess connect via UDP 192.168.17.2:5061 --> 192.168.15.3:5060
ua: using AF from sdp offer: af=AF_INET
call: alloc with params laddr=192.168.15.3, af=AF_INET, use_rtp=1
call: use_video=0
call: no common audio codecs - rejected
audio: destroyed (started=0)

Debugging from my WS to UDP node.js program, which dumps the modified contents of each SIP message it handles after the "Sent to xxx {}" dump which shows where it sent the message that follows.

Sent to UDP: {
  transport: 'UDP',
  isSip: true,
  user: 'peer',
  port: '5060',
  address: '192.168.15.3'
}
INVITE sip:peer@192.168.15.3:5060 SIP/2.0
Via: SIP/2.0/UDP 192.168.17.3:5061;branch=z9hG4bK000000002
Via: SIP/2.0/WS 1o8bti4vmrgn.invalid;branch=z9hG4bK1880269
Max-Forwards: 68
To: <sip:peer@192.168.15.3:5060>
From: <sip:local@invalid>;tag=n3al8beum4
Call-ID: s85mt5mjcib174kg2ace
CSeq: 3556 INVITE
Contact: <sip:oeqe9v5f@1o8bti4vmrgn.invalid;transport=ws;ob>
Content-Type: application/sdp
Session-Expires: 90
Allow: INVITE,ACK,CANCEL,BYE,UPDATE,MESSAGE,OPTIONS,REFER,INFO,NOTIFY
Supported: timer,ice,replaces,outbound
User-Agent: JsSIP 3.10.0
Content-Length: 1576

v=0
o=mozilla...THIS_IS_SDPARTA-99.0 8984607392562573164 0 IN IP4 0.0.0.0
s=-
t=0 0
a=sendrecv
a=fingerprint:sha-256 89:19:DA:4C:69:DD:D6:6A:40:55:26:9F:82:90:1A:E7:EA:77:3A:DB:25:B8:91:F2:20:75:BA:7A:40:89:91:63
a=group:BUNDLE 0
a=ice-options:trickle
a=msid-semantic:WMS *
m=audio 38701 UDP/TLS/RTP/SAVPF 109 9 0 8 101
c=IN IP4 192.168.17.2
a=candidate:0 1 UDP 2122187007 192.168.11.99 36945 typ host
a=candidate:1 1 UDP 2122252543 192.168.17.2 38701 typ host
a=candidate:2 1 TCP 2105458943 192.168.11.99 9 typ host tcptype active
a=candidate:3 1 TCP 2105524479 192.168.17.2 9 typ host tcptype active
a=candidate:0 2 UDP 2122187006 192.168.11.99 33255 typ host
a=candidate:1 2 UDP 2122252542 192.168.17.2 47105 typ host
a=candidate:2 2 TCP 2105458942 192.168.11.99 9 typ host tcptype active
a=candidate:3 2 TCP 2105524478 192.168.17.2 9 typ host tcptype active
a=sendrecv
a=end-of-candidates
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=extmap:2/recvonly urn:ietf:params:rtp-hdrext:csrc-audio-level
a=extmap:3 urn:ietf:params:rtp-hdrext:sdes:mid
a=fmtp:109 maxplaybackrate=48000;stereo=1;useinbandfec=1
a=fmtp:101 0-15
a=ice-pwd:f65f2a844c7400ce28833a19f1e8a6b1
a=ice-ufrag:3430adeb
a=mid:0
a=msid:{bc0512c4-d978-40ba-8530-29503c3c62c7} {1f8888a4-c7e6-4e07-8562-ddbb7e7f465b}
a=rtcp:47105 IN IP4 192.168.17.2
a=rtcp-mux
a=rtpmap:109 opus/48000/2
a=rtpmap:9 G722/8000/1
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:101 telephone-event/8000
a=setup:actpass
a=ssrc:1929120043 cname:{e2873cf2-3f31-41db-8e88-5918bf5d713d}

Sent to WS: { transport: 'WS', port: undefined, address: '1o8bti4vmrgn.invalid' }
SIP/2.0 488 Not Acceptable Here
Via: SIP/2.0/WS 1o8bti4vmrgn.invalid;branch=z9hG4bK1880269
To: <sip:peer@192.168.15.3:5060>;tag=70fa1bcb8b63782c
From: <sip:local@invalid>;tag=n3al8beum4
Call-ID: s85mt5mjcib174kg2ace
CSeq: 3556 INVITE
Server: baresip v1.0.0 (x86_64/linux)
Content-Length: 0

Sent to UDP: {
  transport: 'UDP',
  isSip: true,
  user: 'peer',
  port: '5060',
  address: '192.168.15.3'
}
ACK sip:peer@192.168.15.3:5060 SIP/2.0
Via: SIP/2.0/UDP 192.168.17.3:5061;branch=z9hG4bK000000003
Via: SIP/2.0/WS 1o8bti4vmrgn.invalid;branch=z9hG4bK1880269
Max-Forwards: 68
To: <sip:peer@192.168.15.3:5060>;tag=70fa1bcb8b63782c
From: <sip:local@invalid>;tag=n3al8beum4
Call-ID: s85mt5mjcib174kg2ace
CSeq: 3556 ACK
Allow: INVITE,ACK,CANCEL,BYE,UPDATE,MESSAGE,OPTIONS,REFER,INFO,NOTIFY
Supported: outbound
User-Agent: JsSIP 3.10.0
Content-Length: 0

I don't see anything wrong with the sip messaging which appears to be going back and forth through my bridge program properly, including the ACK from the web client of the "Not Acceptable Here" message from baresip.  Any one have an idea why the call is being rejected with a "no common audio codecs" error?

I found some interesting conversations about the "m=audio 38701 UDP/TLS/RTP/SAVPF 109 9 0 8 101" in the invite message, so I also tried modifying the line in my bridge with this code snippet...

if(line.startsWith("m=audio ")){
	line = line.replace("UDP/TLS/", ""); // strip off WebRTC mangling of standard SIP
	lines[n] = line; // Modified line back in message before sending
}

... and it made no difference.

Any insight?

Thanks,
-Ethan

**Comment by sreimers** [comment]
`$ baresip -v
baresip v1.0.0 Copyright (C) 2010 - 2020 Alfred E. Heggestad et al.`

Baresip v1.0.0 is quite old and outdated, we just released v3.13.0, can you test with latest release?

https://github.com/baresip/baresip/wiki/Install:-Stable-Release

**Comment by eafunk** [comment]
Build and install process worked without errors or warning, but baresip crashes when I try to run the newly version.
This was build from the most recent release, and is running on Ubuntu 24.04.

-Ethan

$ baresip
baresip v3.13.0 Copyright (C) 2010 - 2024 Alfred E. Heggestad et al.
Local network addresses:
     enp1s0:  192.168.15.3
     enp1s0:  fe80::9f4e:5bd1:df9e:b31a
*** stack smashing detected ***: terminated

**Comment by eafunk** [comment]
Reboot of the computer fixed the stack smashing problem. Unfortunately, the exact same problem persists.  Same SIP conversation, same errors on the console.  So this is not looking like an old version problem.  When basesip says "no common audio or video codecs" is that with regard to just codecs, or does that include  other media transport elements outside of the audio codec list itself.  For example, STUN, or lack of STUN in my case?

-Ethan
